crates/async-compression/Cargo.toml

```diff
@@ -32,41 +32,32 @@ all-algorithms = [
 ]
 
 # algorithms
-brotli = ["compression-codecs/brotli", "dep:brotli"]
-bzip2 = ["compression-codecs/bzip2", "dep:bzip2"]
-deflate = ["compression-codecs/deflate", "flate2"]
-deflate64 = ["compression-codecs/deflate64", "dep:deflate64"]
-gzip = ["compression-codecs/gzip", "flate2"]
-lz4 = ["compression-codecs/lz4", "dep:lz4"]
-lzma = ["compression-codecs/lzma", "liblzma"]
+brotli = ["compression-codecs/brotli"]
+bzip2 = ["compression-codecs/bzip2"]
+deflate = ["compression-codecs/deflate"]
+deflate64 = ["compression-codecs/deflate64"]
+gzip = ["compression-codecs/gzip"]
+lz4 = ["compression-codecs/lz4"]
+lzma = ["compression-codecs/lzma"]
 xz = ["compression-codecs/xz", "lzma"]
-xz-parallel = ["compression-codecs/xz-parallel", "xz", "liblzma/parallel"]
+xz-parallel = ["compression-codecs/xz-parallel", "xz"]
 xz2 = ["compression-codecs/xz2", "xz"]
-zlib = ["compression-codecs/zlib", "flate2"]
-zstd = ["compression-codecs/zstd", "libzstd", "zstd-safe"]
-zstdmt = ["compression-codecs/zstdmt", "zstd", "zstd-safe/zstdmt"]
+zlib = ["compression-codecs/zlib"]
+zstd = ["compression-codecs/zstd"]
+zstdmt = ["compression-codecs/zstdmt", "zstd"]
 
 
 [dependencies]
 # core dependencies
 futures-core.workspace = true
-memchr.workspace = true
 pin-project-lite.workspace = true
 compression-codecs.workspace = true
 compression-core.workspace = true
-# optionals deps
-brotli = { version = "8", optional = true }
-bzip2 = { version = "0.6", optional = true }
-flate2 = { version = "1.0.13", optional = true }
```

Comment:
technically this removes a feature, but in reality this feature does nothing and should never be used to begin with (should have marked them as `dep:` but)